### DISCUSSION TITLE
Possible updating library.xml?

### DISCUSSION THREAD

**Discussion by jojo2357** [jojo2357/kiwix-zim-updater#27] [Ideas]
@DocDrydenn I noticed in the readme

> After it's complete, I follow it up with an automated call to update my `library.xml` for the Kiwix server (Note: this part is not provided via this script). This keeps my ZIM library and Kiwix server updated.

which sounds really cool and I want it. could you provide this?

P.S. I would also be interested in adding it to the script, but that might be a challenge.

**Comment by DocDrydenn** [comment]
So... I run Kiwix-Server (+web frontend) as a service in a plain old Debian LXC from my Proxmox server. 

It's run via `ExecStart=/usr/bin/kiwix-serve --port 8080 --library /path/to/library.xml /path/to/zims/`, so before you can launch it, you must generate (or already have) a `library.xml`.

A `library.xml` file can be generated by running `kiwix-manage /path/to/library.xml add /path/to/zims/*.zim`. It scans the zim path for all zims and adds them to the `library.xml` file. It's output is just the `library.xml` file in the location specified.

Now, anytime you change/modify the ZIM directory, you have to re-run the `library.xml` generation command. It'll overwrite whatever `library.xml` might already be there, so there's no need to delete the old one first.

Finally, in order for the Kiwix-Server service to read the new `library.xml`, you just have to restart the service.
